Source/UnitTests/Core/PowerPC/JitArm64/MovI2R.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Common/Arm64Emitter.h"
+#include "Common/Random.h"
+
+#include <fmt/format.h>
+#include <gtest/gtest.h>
+
+namespace
+{
+using namespace Arm64Gen;
+
+class TestMovI2R : public ARM64CodeBlock
+{
+public:
+  TestMovI2R() { AllocCodeSpace(4096); }
+
+  void Check32(u32 value)
+  {
+    ResetCodePtr();
+
+    auto fn = (u32(*)())GetCodePtr();
+    MOVI2R(W0, value);
+    RET();
+
+    FlushIcacheSection((u8*)fn, (u8*)GetCodePtr());
+
+    EXPECT_EQ(value, fn());
+  }
+
+  void Check64(u64 value)
+  {
+    ResetCodePtr();
+
+    auto fn = (u64(*)())GetCodePtr();
+    MOVI2R(X0, value);
+    RET();
+
+    FlushIcacheSection((u8*)fn, (u8*)GetCodePtr());
+
+    EXPECT_EQ(value, fn());
+  }
+};
+
+}  // namespace
+
+TEST(JitArm64, MovI2R_32BitValues)
+{
+  TestMovI2R test;
+  for (u64 i = 0; i < 0xFFFFFFFF; i++)
+  {
+    test.Check32((u32)i);
+    test.Check64(i);
+  }
+}
+
+TEST(JitArm64, MovI2R_Rand)
+{
+  TestMovI2R test;
+  for (u64 i = 0; i < 0x100000; i++)
+  {
+    test.Check64(Common::Random::GenerateValue<u64>());
```

Comment:
For reproducibility (e.g. if you're trying to fix a bug and you want to be sure that the reason the tests started passing actually is because your fix works), I think it would be nice to use a PRNG that takes a constant seed. I don't think we have any in `Common`, but the `<random>` header has some.